pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -19,12 +19,207 @@ package kuberuntime
+			if config.Linux == nil {
+				config.Linux = &runtimeApi.LinuxContainerConfig{}
+			}
+			config.Linux.SelinuxOptions = &runtimeApi.SELinuxOption{
+				User:  &securityContext.SELinuxOptions.User,
+				Role:  &securityContext.SELinuxOptions.Role,
+				Type:  &securityContext.SELinuxOptions.Type,
+				Level: &securityContext.SELinuxOptions.Level,
+			}
+		}
+	}
+
+	envs := make([]*runtimeApi.KeyValue, len(opts.Envs))
+	for index, e := range opts.Envs {
+		envs[index] = &runtimeApi.KeyValue{
+			Key:   &e.Name,
+			Value: &e.Value,
+		}
+	}
+	config.Envs = envs
+
+	return config, nil
+}
+
+// makeMounts generates container volume mounts for kubelet runtime api.
+func makeMounts(cid string, opts *kubecontainer.RunContainerOptions, container *api.Container, podHasSELinuxLabel bool) []*runtimeApi.Mount {
+	volumeMounts := []*runtimeApi.Mount{}
+
+	for _, v := range opts.Mounts {
+		m := &runtimeApi.Mount{
+			Name:          &v.Name,
+			HostPath:      &v.HostPath,
+			ContainerPath: &v.ContainerPath,
+			Readonly:      &v.ReadOnly,
+		}
+		if podHasSELinuxLabel && v.SELinuxRelabel {
+			m.SelinuxRelabel = &v.SELinuxRelabel
+		}
+
+		volumeMounts = append(volumeMounts, m)
+	}
+
+	// The reason we create and mount the log file in here (not in kubelet) is because
+	// the file's location depends on the ID of the container, and we need to create and
+	// mount the file before actually starting the container.
+	if opts.PodContainerDir != "" && len(container.TerminationMessagePath) != 0 {
+		// Because the PodContainerDir contains pod uid and container name which is unique enough,
+		// here we just add an unique container id to make the path unique for different instances
+		// of the same container.
+		containerLogPath := path.Join(opts.PodContainerDir, cid)
+		fs, err := os.Create(containerLogPath)
+		if err != nil {
+			glog.Errorf("Error on creating termination-log file %q: %v", containerLogPath, err)
+		} else {
+			fs.Close()
+			volumeMounts = append(volumeMounts, &runtimeApi.Mount{
+				HostPath:      &containerLogPath,
+				ContainerPath: &container.TerminationMessagePath,
+			})
+		}
+	}
+
+	return volumeMounts
+}
+
+// getKubeletContainers lists all (or just the running) containers managed by kubelet.
+func (m *kubeGenericRuntimeManager) getKubeletContainers(allContainers bool) ([]*runtimeApi.Container, error) {
+	var resp []*runtimeApi.Container
+	var err error
+
+	if allContainers {
+		resp, err = m.runtimeService.ListContainers(nil)
+	} else {
+		runningState := runtimeApi.ContainerState_RUNNING
+		resp, err = m.runtimeService.ListContainers(&runtimeApi.ContainerFilter{
+			State: &runningState,
+		})
+	}
+	if err != nil {
+		glog.Errorf("ListContainers failed: %v", err)
+		return nil, err
+	}
+
+	result := []*runtimeApi.Container{}
+	for _, c := range resp {
+		if len(c.GetName()) == 0 {
+			continue
+		}
+
+		if !isContainerManagedByKubelet(c.GetName()) {
+			glog.V(3).Infof("Container %s is not managed by kubelet", c.GetName())
```

Comment:
I know it's V(3) in the existing code, but periodically printing this hasn't be useful to me. Can we bump this to V(5)?